tests/scripts/all.sh

```diff
@@ -438,6 +438,16 @@ if uname -a | grep -F x86_64 >/dev/null; then
 msg "build: i386, make, gcc" # ~ 30s
 cleanup
 CC=gcc CFLAGS='-Werror -Wall -Wextra -m32' make
+
+msg "test: i386, make, gcc"
+make test
+
+msg "build: 64-bit ILP32, make, gcc" # ~ 30s
+cleanup
+CC=gcc CFLAGS='-Werror -Wall -Wextra -mx32' make
```

Comment:
In what environment did you test this? On my Ubuntu 16.04 amd64 PC, the build works with `-m32`, but not with `-mx32`. It works with `-mx32 -I/usr/include/i386-linux-gnu` but I don't think that's the right way to do it, and anyway whatever's in the script needs to work in our reference environment. I have `gcc-5-multilib`, `libc6-dev-x32` and `libc6-dev-i386` and possibly other relevant packages installed, what could I be missing?

Because of this I am not merging this PR at this point. It looks good otherwise.

I've started a CI run of `all.sh`: https://jenkins-internal.mbed.com/job/mbedtls-release/44/

Reply:
Sorry, I forgot to quote the error. It's due to the attempt to call a system call under Linux.

```
~/z/dev/mbedtls/misc % make CFLAGS=-mx32          
    …
  CC    entropy_poll.c
In file included from entropy_poll.c:92:0:
/usr/include/sys/syscall.h:24:24: fatal error: asm/unistd.h: No such file or directory
compilation terminated.
```

Reply:
@gilles-peskine-arm: This test works in my machine, but I am not sure which package its missing in the test environment. I have the following installed:
```
g++-5-multilib/xenial-security,now 5.4.0-6ubuntu1~16.04.4 amd64 [installed,upgradable to: 5.4.0-6ubuntu1~16.04.5] 
g++-multilib/xenial,now 4:5.3.1-1ubuntu1 amd64 [installed]                          
gcc-5-multilib/xenial-security,now 5.4.0-6ubuntu1~16.04.4 amd64 [installed,upgradable to: 5.4.0-6ubuntu1~16.04.5]                            
gcc-multilib/xenial,now 4:5.3.1-1ubuntu1 amd64 [installed,automatic]  

libc6-dev-i386/xenial-updates,xenial-security,now 2.23-0ubuntu9 amd64 [installed,automatic]
libc6-dev-x32/xenial-updates,xenial-security,now 2.23-0ubuntu9 amd64 [installed,automatic]
libc6-i386/xenial-updates,xenial-security,now 2.23-0ubuntu9 amd64 [installed,automatic]
libc6-x32/xenial-updates,xenial-security,now 2.23-0ubuntu9 amd64 [installed,automatic]
```

Reply:
I have all of these except `gcc-multilib` and `g++-multilib`, but they're dependency-only packages and I have all their dependencies.

Reply:
@gilles-peskine-arm: How about these?
```
libx32gcc-5-dev/xenial-security,now 5.4.0-6ubuntu1~16.04.4 amd64 [installed,upgradable to: 5.4.0-6ubuntu1~16.04.5]
libx32gcc1/xenial,now 1:6.0.1-0ubuntu1 amd64 [installed,automatic]              
libx32stdc++-5-dev/xenial-security,now 5.4.0-6ubuntu1~16.04.4 amd64 [installed,upgradable to: 5.4.0-6ubuntu1~16.04.5]
libx32stdc++6/xenial-security,now 5.4.0-6ubuntu1~16.04.4 amd64 [installed,upgradable to: 5.4.0-6ubuntu1~16.04.5]
```

Reply:
I have those as well.

I wonder if it could be a conflicting package. But I don't see what it could be.

Reply:
Got it! `gcc-multilib` is *amost* a dependency-only package, but it creates one crucial symbolic link:

    ln -s x86_64-linux-gnu/asm /usr/include/asm

Works for me now.

Reply:
@gilles-peskine-arm: Thanks for taking a look! Does this unblock the merge?

Reply:
Thanks for your help. `all.sh` has run on CI on the PR. Since there's a merge conflict, even if it's a trivial one, I'm running `all.sh` on the merge result locally and will push if successful.